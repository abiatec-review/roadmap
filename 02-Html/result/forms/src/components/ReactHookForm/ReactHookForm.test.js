import {fireEvent, render, screen} from "@testing-library/react";
import ReactHookForm from "./ReactHookForm";
import '@testing-library/jest-dom'
import userEvent from "@testing-*//user-event";

describe("App", () => {
    it('renders ReackHookForm component', () => {
        render(<ReactHookForm />)

        expect(screen.getByText(/Banana/i)).toBeInTheDocument()
        expect(screen.getByText(/Choose fruits you like/i)).toBeInTheDocument()

        expect(screen.queryByText(/Choose your favorite anime/i)).toBeNull()
        expect(screen.queryByText(/Choose fruits you like/i)).toHaveClass('h3-element')
    })

    it('check click on checkbox', () => {
        render(<ReactHookForm />)

        const label = screen.getByText(/Banana/i)
        const input = screen.getByTestId('first-input')
        userEvent.click(label)

        expect(input).toBeChecked()
    })
    it('check if error message works correctly', async() => {
        render(<ReactHookForm />)

        const button = screen.queryByTestId('btn')
        expect(screen.queryByTestId('error-text')).toBeNull()
        fireEvent.click(button)

        expect(await screen.findByText('fruits field must have at least 1 items'));
    })
    it('check if next questions appear correctly', async() => {
        render(<ReactHookForm />)

        const button = screen.queryByTestId('btn')
        const label = screen.getByText(/Banana/i)
        fireEvent.click(label)
        fireEvent.click(button)

        expect(await screen.findByText('Naruto'));
    })
})