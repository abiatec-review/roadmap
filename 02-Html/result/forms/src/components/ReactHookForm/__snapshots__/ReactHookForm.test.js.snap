// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <section
    class="firstQ"
  >
    <h3>
      Choose fruits you like
    </h3>
    <form
      class="form"
    >
      <label>
        Banana
        <input
          name="fruits"
          type="checkbox"
          value="Banana"
        />
      </label>
      <label>
        Apple
        <input
          name="fruits"
          type="checkbox"
          value="Apple"
        />
      </label>
      <label>
        Watermelon
        <input
          name="fruits"
          type="checkbox"
          value="Watermelon"
        />
      </label>
      <label>
        Pineapple
        <input
          name="fruits"
          type="checkbox"
          value="Pineapple"
        />
      </label>
      <label>
        Pear
        <input
          name="fruits"
          type="checkbox"
          value="Pear"
        />
      </label>
      <button
        class="undefined"
      >
        Next question
      </button>
    </form>
  </section>
</DocumentFragment>
`;
